### ISSUE TITLE
Steal 30% of your ad! 

### ISSUE THREAD

**Issue by InkOut** [floatinghotpot/cordova-admob-pro#544]
I thought that i give you only 2% [https://github.com/floatinghotpot/cordova-admob-pro/wiki/License-Agreement#3-win-win-partnership](url), but i checked url and saw, that 2% is 30%.

Plugin make a request to "http://adlic.rjfun.com/adlic" and get a JSON with the "%"

**Comment by haidang1904** [comment]
this is a fact??? OMG. I have implement to many my game app. this is first time I just free/ but not free plugin. any body can explain clean this issues for me. I can't rebuild all my game app just to remove this plugin. when I start using this pluggin. I suspect something is not right. my ads show with plugin slow than other my project (not using cordova). some time my ads not show. but show on other project???
so, my question:
 1. how many% of ads are stolen???
 2. what happen if this plugin cannot get request to  "http://adlic.rjfun.com/adlic"? my ads stop working?
 3. any soluation/plugin "FREE" for cordova to replace this plugin. 
@@this is bad day

**Comment by InkOut** [comment]
@floatinghotpot speek about 2% but it is 30. My revenue from 1 jan to today is 4 371,80$, so @floatinghotpot stole 1873$

**Comment by haidang1904** [comment]
@InkOut thanks you for your info. how to check % on this side. I want to check on my app. thanks you again for your fast reply :(

**Comment by InkOut** [comment]
But i use this plugin from 1 september 2016. More then 4500$ has been stolen

You can download https://www.charlesproxy.com/ and look to your data

**Comment by haidang1904** [comment]
OMG. I can't belive this is Free plugin. No any this info on wiki or cordova/plugin site. any plugin can replace this? @InkOut thanks you again for your.

**Comment by InkOut** [comment]
@haidang1904 `curl --request POST \
  --url http://adlic.rjfun.com/adlic \
  --header 'content-type: multipart/form-data;' \
  --form app=ru.inkout.mcko \
  --form os=ios \
  --form net=admob`

run it in terminal. "r" is %

**Comment by InkOut** [comment]
@haidang1904 i change this plugin to https://github.com/ratson/cordova-plugin-admob-free
And i want to return my money.

4500$ is really a lot of

**Comment by InkOut** [comment]
request to `curl --request POST --url http://adlic.rjfun.com/adlic --header 'content-type: multipart/form-data;' --form app=ru.inkout.mcko --form os=ios --form net=admob`
returns 
`{"r":30,"b":"ca-app-pub-9389363019208436/1839284301","i":"ca-app-pub-9389363019208436/3316017508","n":"","v":"ca-app-pub-9389363019208436/4792750707","js":""}`

**Comment by InkOut** [comment]
It is license:

> If use in commercial project, please get a license, or, you have monetized more than $1000 using this plugin, you are also required to: either get a commercial license ($20), or, no need pay but share 2% Ad traffic, read the License Agreement for details

Nothing about 30%. 20$ or 2%. 2% is ok, but 30%...

**Comment by haidang1904** [comment]
@InkOut thanks for your link 
I just replace my ID below right?
curl --request POST --url http://adlic.rjfun.com/adlic --header 'content-type: multipart/form-data;' --form app=# replace.this.right --form os=ios --form net=admob

**Comment by InkOut** [comment]
@haidang1904 yes. Send me your id plz

**Comment by haidang1904** [comment]
@InkOut any contact info with you. please add me haidang1904@gmail.com gmail/skype/anything

**Comment by InkOut** [comment]
Can someone help me with court? 
How i can return money if @floatinghotpot dont return

@haidang1904 i sant you an email

**Comment by MoAsmar** [comment]
@InkOut  I notice your issue yesterday, I think the License agreement itself is not very clear, as it does not say if the plugin take by default 2% of your Ad Traffic as soon as you pass the 1000$ limit, is it based on trust that you will say this happened, or it is implemented in the plugin already, if so anyone can change the code as it is an open source project and such a cheat cannot be hidden.
Let's give the developer a chance to reply, then we decide. I am using this plugin too from a long time ago more than 2 years.

**Comment by InkOut** [comment]
Still no answer from @floatinghotpot.
And he still steal 30%
Check it: `curl --request POST --url http://adlic.rjfun.com/adlic --header 'content-type: multipart/form-data;' --form app=ru.inkout.mcko --form os=ios --form net=admob`

**Comment by lordamon** [comment]
I'm a paying user, i really want some feedback from @floatinghotpot about this issue.

**Comment by haidang1904** [comment]
Please direct contact with him via: 
rjfun.mobile@gmail.com
linkedin.com/in/limingxie
Twitter: limingxie @rjfun 

thanks @InkOut again for this topic. Please stop using products "make by chinese"

**Comment by floatinghotpot** [comment]
@InkOut sorry for a late response, I just saw your email, and already replied.

Similar issue happened before, and we already explained in these post: 
License Agreement - https://github.com/floatinghotpot/cordova-admob-pro/wiki/License-Agreement
#450 - "Not only 2% of Sharing ad traffic?"
#470 - "More cost effective to get a commercial license if high traffic".

If any further questions, please drop us email for further communication: rjfun.mobile@gmail.com

**Comment by floatinghotpot** [comment]
@InkOut sorry for a late response, I just saw your email and already replied.

Some developers raise the issue before, and here are some posts related for your reference.
#450 - "Not only 2% of Sharing ad traffic?"
#470 - "More cost effective to get a commercial license if high traffic"

Please send us your app id via email (rjfun.mobile@gmail.com), and will double check it for you. If what your said is true, we will calculate, deduct the license fee, and return the rest.

Again, please read it carefully:
License Agreement - https://github.com/floatinghotpot/cordova-admob-pro/wiki/License-Agreement

**Comment by InkOut** [comment]
@floatinghotpot my app id: ru.inkout.mcko

**Comment by InkOut** [comment]
Status update.

Good news: @floatinghotpot agree to return money! Its good!
Now we are discussing how much money he will return.

Problem:
This plugin have anti-hack system. 
> Usually when a guy is using a fake license key, or send unusual attacking request (for example, request not from iOS/android cordova platform), the ant-crack logic will be triggered and move the app into the black list.

> You may not know that, for any new app, the ad sharing is 0% from beginning, not 2%. When the system calculate your earning exceed $1000, if no license key is still provided, it will start to applied 2% ad sharing. But, when unusual request detected, it will increase the rate bit by bit, but no more than 30%. So your rough guess on shared revenue is not correct.

So, i am sure that @floatinghotpot will do something with this problem, because no one should be responsible for the actions of hackers.
I am sure @floatinghotpot is not trying to steal money, because he was agreed to return the money.

**Comment by InkOut** [comment]
@floatinghotpot is really good guy! I cant believe that it is true, but he already returned half of the amount!

**Comment by floatinghotpot** [comment]
Many guys pay $20 to get a license, then 100% use their own ad unit id, and no ad sharing applied to their apps at all. Our customers are very satisfied with our prompt support. 

We did find there are some guys using fake license, will trigger the ant-crack logic. And, a few guys sent simulated requests from PC may also trigger it. So, it’s not wise to attempt to access the license server directly.

Thanks for discussing with us in email, the communication is effective. We trust @InkOut did not attempt to hack intentionally. So we keep 2% as we stated in license agreement, and have returned the rest to @InkOut via PayPal.
